test/cpp/c10d/ProcessGroupNCCLErrorsTest.cpp

```diff
@@ -346,23 +347,64 @@ TEST_F(ProcessGroupNCCLErrorsTest, testNCCLErrorsNonBlocking) {
   // Communicators might be aborted here, further operations would fail.
 }
 
+// Function to read what we wrote to the local disk for validation.
+std::string readTraceInfoFromLocalDisk(int rank) {
+  auto filename = c10::str("/tmp/test_nccl_trace_rank_", rank);
```

Comment:
doesnt the filename also need be read from the `TORCH_NCCL_DEBUG_INFO_TEMP_FILE` env variable?

Reply:
I think for testing we can use what Howard suggests -- `tmpfile`.

Reply:
@H-Huang yes, for test purpose, I just skipped it. I will added it in the test. And I will use `tmpfile` here @XilunWu